src/test/java/edu/ucsb/cs156/example/controllers/PersonalSchedulesControllerTests.java

```diff
@@ -0,0 +1,293 @@
+                .andExpect(status().is(403));
+    }
+
+    @WithMockUser(roles = { "USER" })
+    @Test
+    public void api_schedules_all__user_logged_in__returns_200() throws Exception {
+        mockMvc.perform(get("/api/personalschedules/all"))
+                .andExpect(status().isOk());
+    }
+
+    // Authorization tests for /api/personalschedules/post
+
+    @Test
+    public void api_schedules_post__logged_out__returns_403() throws Exception {
+        mockMvc.perform(post("/api/personalschedules/post"))
+                .andExpect(status().is(403));
+    }
+
+    // Tests with mocks for database actions
+
+    @WithMockUser(roles = { "USER" })
+    @Test
+    public void api_schedules__user_logged_in__returns_a_schedules_that_exists() throws Exception {
+
+        // arrange
+
+        User u = currentUserService.getCurrentUser().getUser();
+        PersonalSchedule personalschedule1 = PersonalSchedule.builder().name("Name 1").description("Description 1").quarter("Quarter 1").user(u).id(7L).build();
+        when(personalscheduleRepository.findByIdAndUser(eq(7L), eq(u))).thenReturn(Optional.of(personalschedule1));
+
+        // act
+        MvcResult response = mockMvc.perform(get("/api/personalschedules?id=7"))
+                .andExpect(status().isOk()).andReturn();
+
+        // assert
+
+        verify(personalscheduleRepository, times(1)).findByIdAndUser(7L, u);
+        String expectedJson = mapper.writeValueAsString(personalschedule1);
+        String responseString = response.getResponse().getContentAsString();
+        assertEquals(expectedJson, responseString);
+    }
+
+    @WithMockUser(roles = { "USER" })
+    @Test
+    public void api_schedules__user_logged_in__search_for_schedules_that_does_not_exist() throws Exception {
+
+        // arrange
+
+        User u = currentUserService.getCurrentUser().getUser();
+
+        when(personalscheduleRepository.findByIdAndUser(eq(7L), eq(u))).thenReturn(Optional.empty());
+
+        // act
+        MvcResult response = mockMvc.perform(get("/api/personalschedules?id=7"))
+                .andExpect(status().isNotFound()).andReturn();
+
+        // assert
+
+        verify(personalscheduleRepository, times(1)).findByIdAndUser(7L, u);
+        Map<String, Object> json = responseToJson(response);
+        assertEquals("EntityNotFoundException", json.get("type"));
+        assertEquals("PersonalSchedule with id 7 not found", json.get("message"));
+    }
+
+    @WithMockUser(roles = { "USER" })
+    @Test
+    public void api_schedules__user_logged_in__search_for_schedules_that_belongs_to_another_user() throws Exception {
+
+        // arrange
+
+        User u = currentUserService.getCurrentUser().getUser();
+        User otherUser = User.builder().id(999L).build();
+        PersonalSchedule otherUsersPersonalschedule = PersonalSchedule.builder().name("Name 1").description("Description 1").quarter("Quarter 1").user(otherUser).id(13L)
```

Comment:
Quarter shoud be in YYYYQ format, e.g. 20224 for F22, 20192 for S19, etc.

So instead:

```suggestion
PersonalSchedule otherUsersPersonalschedule = PersonalSchedule.builder().name("Name 1").description("Description 1").quarter("20222").user(otherUser).id(13L)
```